src/CMakeLists.txt

```diff
@@ -155,100 +155,88 @@ if(WITHOUT_WIN32_CONFIG)
   endif(WITH_PLUGINS)
 endif()
 
-if(RDKAFKA_BUILD_STATIC)
-  set(CMAKE_POSITION_INDEPENDENT_CODE ON)
-  set(RDKAFKA_BUILD_MODE STATIC)
-else()
-  set(RDKAFKA_BUILD_MODE SHARED)
-endif()
+set(CMAKE_DEBUG_POSTFIX d)
 
-add_library(rdkafka ${RDKAFKA_BUILD_MODE} ${sources})
-if(NOT RDKAFKA_BUILD_STATIC)
-  set_property(TARGET rdkafka PROPERTY SOVERSION ${LIBVER})
-endif()
+add_library(rdkafka SHARED ${sources})
+add_library(rdkafka_static STATIC ${sources})
+set_property(TARGET rdkafka PROPERTY SOVERSION ${LIBVER})
+target_compile_definitions(rdkafka PRIVATE LIBRDKAFKA_EXPORTS)
+target_compile_definitions(rdkafka_static PUBLIC LIBRDKAFKA_STATICLIB)
 
 if(MINGW)
   # Target Windows 8.1 to match the VS projects (MinGW defaults to an older WinAPI version)
   list(APPEND rdkafka_compile_definitions WINVER=0x0603 _WIN32_WINNT=0x0603 UNICODE)
 endif(MINGW)
 
-# Support '#include <rdkafka.h>'
-target_include_directories(rdkafka PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}>")
-target_compile_definitions(rdkafka PUBLIC ${rdkafka_compile_definitions})
-if(RDKAFKA_BUILD_STATIC)
-  target_compile_definitions(rdkafka PUBLIC LIBRDKAFKA_STATICLIB)
-endif()
-
-if(MINGW)
-	message(STATUS "Detected MinGW build")
-  # MinGW ignores `#pragma comment(lib, ...)` so we need to specify link libs
-  target_link_libraries(rdkafka PUBLIC crypt32 ws2_32 secur32)
-endif(MINGW)
-
-if(WIN32)  
-  if(RDKAFKA_BUILD_STATIC)
-    target_link_libraries(rdkafka PUBLIC crypt32)
-  else()
-    target_compile_definitions(rdkafka PRIVATE LIBRDKAFKA_EXPORTS)
-  endif()
-endif()
-
 # We need 'dummy' directory to support `#include "../config.h"` path
 set(dummy "${GENERATED_DIR}/dummy")
 file(MAKE_DIRECTORY "${dummy}")
-target_include_directories(rdkafka PUBLIC "$<BUILD_INTERFACE:${dummy}>")
 
-if(WITH_HDRHISTOGRAM)
-  target_link_libraries(rdkafka PUBLIC m)
-endif()
+foreach(target rdkafka rdkafka_static)
 
-if(WITH_ZLIB)
-  find_package(ZLIB REQUIRED)
-  target_include_directories(rdkafka PUBLIC ${ZLIB_INCLUDE_DIRS})
-  target_link_libraries(rdkafka PUBLIC ZLIB::ZLIB)
-endif()
+  # Support '#include <rdkafka.h>'
```

Comment:
Why without the directory prefix? librdkafka provides multiple header files and as such should have its own include directory.

Reply:
That comment was actually moved from line 175 on master